types/test.tsx

```diff
@@ -20,6 +23,41 @@ x('urlset', {xmlns}, x('loc'), x('loc')) // $ExpectType Element
 x('urlset', {xmlns}, [x('loc'), x('loc')]) // $ExpectType Element
 x('urlset', {xmlns}, []) // $ExpectType Element
 
+// $ExpectType Element
+const jsx0 = <urlset />
+// $ExpectType Element
+const jsx1 = <urlset xmlns={xmlns} />
+// $ExpectType Element
+const jsx2 = <urlset>string</urlset>
+// $ExpectType Element
+const jsx3 = <urlset>{['string', 'string']}</urlset>
+// $ExpectType Element
+const jsx4 = (
+  <urlset>
+    <loc />
+    string
+  </urlset>
+)
+// $ExpectType Element
+const jsx5 = (
+  <urlset>
+    <loc />
+  </urlset>
+)
+// $ExpectType Element
+const jsx6 = (
+  <urlset>
+    <loc />
+    <loc />
+  </urlset>
+)
+// $ExpectType Element
+const jsx7 = <urlset>{[<loc />, <loc />]}</urlset>
+// $ExpectType Element
+const jsx8 = <urlset>{[]}</urlset>
+
+declare function Bar(props?: Attributes): Element
+const bar = <Bar /> // $ExpectError
```

Comment:
It could be good to have both a ts and a tsx test file, to allow the plain ts features to be tested independent of the tsx syntax.
I looked into this in #7, either this PR or that could move forward, I wanted to avoid committing over @remcohaszing in case there is a difference in opinion.

Reply:
I split it as you requested. :smile: